sirepo/template/flash.py

```diff
@@ -407,15 +406,23 @@ def _cell_size(f, refine_max):
     assert False, 'no blocks with appropriate refine level'
 
 
-def _extract_zip(data, run_dir):
-    import os
-    import sirepo.pkcli.flash
-    import stat
-    import zipfile
-    zipfile.ZipFile(run_dir.join(_SIM_DATA.proprietary_lib_file_basename(data))).extractall()
-    # extractall() doesn't maintain file permissions
-    # https://bugs.python.org/issue15795
-    os.chmod(run_dir.join(_SIM_DATA.EXE_NAME), stat.S_IXUSR)
+def _extract_rpm(data):
+    import distutils.spawn
+    import subprocess
+
+    if distutils.spawn.find_executable(_SIM_DATA.flash_exe_name(data)):
+        return
+    #SECURITY: No user defined input in cmd so shell=True is ok
+    # TODO(e-carlin): think about stdout and stderr, where should they go?
+    subprocess.check_call(
+        'rpm2cpio {} | cpio -idv'.format(
```

Comment:
It's good style to do this:

```py
"rpm2cpio '{}' | cpio --extract --make-directories".format(
```

The quotes ensure that any specials in the file name are not interpreted by the shell. Since this is an internal file, we can assume that the file doesn't have quotes (and probably no shell specials), but people copy this type of code, so it's best to keep to good "shell form".

When passing flags to unusual commands, it's nice to list their gnu long forms.

Reply:
https://github.com/radiasoft/sirepo/pull/2490/commits/00903ca4af2b437c4de426a53cfab1f8b79638eb